src/main/scala-2.12/chisel3/util/experimental/decode/DecoderBundle.scala

```diff
@@ -0,0 +1,22 @@
+package chisel3.util.experimental.decode
+
+import chisel3._
+import scala.collection.immutable.ListMap
+
+/**
+  * Output of DecoderTable
+  * @param fields all fields to be decoded
+  */
+class DecodeBundle(fields: Seq[DecodeField[_, _]]) extends Record with experimental.AutoCloneType {
+  require(fields.map(_.name).distinct.size == fields.size, "Field names must be unique")
+  val elements: ListMap[String, Data] = ListMap(fields.map(k => k.name -> UInt(k.width.W)): _*)
+
```

Comment:
There should be no reason to split between 2.12 and 2.13 here, just use `SeqMap` which we have provided a polyfill for: https://github.com/chipsalliance/chisel3/blob/master/core/src/main/scala-2.12/scala/collection/immutable/package.scala

Reply:
> I believe this is my "[simple work around](https://github.com/chipsalliance/chisel3/pull/2328#issuecomment-1009155565)" which mentioned last year, cc @carlosedp
> And this PR will close #2328 as well...
> Time flies......

@sequencer It seems to me that `apply` method would work for both 2.12 and 2.13 while `from` and `to` methods not. So this is the way I would like to do this.